wallet-new/src/Cardano/Wallet/Types/UtxoStatistics.hs

```diff
@@ -0,0 +1,219 @@
+        ErrAllStakesNegative ->
+            bprint "Utxo statistics allStakes has to be nonnegative"
+
+
+instance ToSchema UtxoStatistics where
+    declareNamedSchema _ = do
+        wordRef <- declareSchemaRef (Proxy :: Proxy Word64)
+        pure $ NamedSchema (Just "UtxoStatistics") $ mempty
+            & type_ .~ SwaggerObject
+            & required .~ ["histogram", "allStakes"]
+            & properties .~ (mempty
+                & at "histogram" ?~ Inline (mempty
+                    & type_ .~ SwaggerObject
+                    & properties .~ (mempty
+                                     & at "10" ?~ wordRef
+                                     & at "100" ?~ wordRef
+                                     & at "1000" ?~ wordRef
+                                     & at "10000" ?~ wordRef
+                                     & at "100000" ?~ wordRef
+                                     & at "1000000" ?~ wordRef
+                                     & at "10000000" ?~ wordRef
+                                     & at "100000000" ?~ wordRef
+                                     & at "1000000000" ?~ wordRef
+                                     & at "10000000000" ?~ wordRef
+                                     & at "100000000000" ?~ wordRef
+                                     & at "1000000000000" ?~ wordRef
+                                     & at "10000000000000" ?~ wordRef
+                                     & at "100000000000000" ?~ wordRef
+                                     & at "1000000000000000" ?~ wordRef
+                                     & at "10000000000000000" ?~ wordRef
+                                     & at "45000000000000000" ?~ wordRef
+                                    )
+                )
+                & at "allStakes" ?~ (Inline $ mempty
+                    & type_ .~ SwaggerNumber
+                    & minimum_ .~ Just 0
+                )
+            )
+
+instance Arbitrary UtxoStatistics where
+    arbitrary = UtxoStatistics <$> arbitrary
+                               <*> arbitrary
+
+instance Buildable [HistogramBar] where
+    build =
+        bprint listJson
+
+
+deriveSafeBuildable ''UtxoStatistics
+instance BuildableSafeGen UtxoStatistics where
+    buildSafeGen _ UtxoStatistics{..} = bprint ("{"
+        %" histogram="%build
+        %" allStakes="%build
+        %" }")
+        theHistogram
+        theAllStakes
+
+instance Example HistogramBar
+instance Example UtxoStatistics
+
+
+computeUtxoStatistics :: [Word64] ->  UtxoStatistics
+computeUtxoStatistics xs = L.fold (summarizeUtxoStatistics $ generateBounds Log10) xs
+
+--  Using foldl library enable as to capture a number of aggregations in one pass. This thanks to L.Fold being an Applicative
+summarizeUtxoStatistics :: NonEmpty Word64 -> L.Fold Word64 UtxoStatistics
+summarizeUtxoStatistics bounds =
+    UtxoStatistics
+    <$> populateBuckets bounds
+    <*> L.sum
```

Comment:
A bit the same as below, minor styling remark but we could avoid the extra `summarizeUtxoStatistics` and make the implementation a bit more straightforward: 

```hs
computeUtxoStatistics :: [Word64] ->  UtxoStatistics
computeUtxoStatistics = L.fold $ UtxoStatistics
    <$> foldBuckets (generateBounds Log10)
    <*> L.sum
```

Reply:
done